clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetch.java

```diff
@@ -95,6 +102,13 @@ public int numRecords() {
         return numRecords;
     }
 
+    /**
+     * @return the next offsets and metadata (last epochs is included)
+     */
+    public Map<TopicPartition, OffsetAndMetadata> nextOffsetAndMetadata() {
```

Comment:
I'd probably just call this method `nextOffsets` to match what you have in `ConsumerRecords.nextOffsets`. You can also copy the description from that method with identical words. Seems like less effort.